### PR TITLE
irmin: Portable in tree

### PR DESCRIPTION
**Pull request by Ngoguey42**
On top of #1749

### CODE REVIEW

File: src/irmin/tree_intf.ml [hunk 1]
```diff
@@ -97,7 +97,13 @@ module type S = sig
   (** The exception raised by functions that attempt to load {!pruned} tree
       nodes. *)
 
-  type error = [ `Dangling_hash of hash | `Pruned_hash of hash ]
+  exception Portable_value of { context : string }
+  (** The exception raised by functions that attemps to perform IO on a portable
```
Comment (CraigFe) [inline src/irmin/tree_intf.ml:101]:
```suggestion
  (** The exception raised by functions that attempts to perform IO on a portable
```

File: src/irmin/tree_intf.ml [hunk 2]
```diff
@@ -363,7 +369,10 @@ module type S = sig
   val counters : unit -> counters
   val dump_counters : unit Fmt.t
   val reset_counters : unit -> unit
-  val inspect : t -> [ `Contents | `Node of [ `Map | `Key | `Value | `Pruned ] ]
+
+  val inspect :
+    t ->
+    [ `Contents | `Node of [ `Map | `Key | `Value | `Portable_dirty | `Pruned ] ]
```
Comment (CraigFe) [inline src/irmin/tree_intf.ml:375]: Probably worth documenting these at some point, especially if they're going to appear in exception messages when interacting with trees.

**Review by CraigFe — APPROVED** [review]

File: src/irmin/tree_intf.ml (hunk 2 above)
Comment (Ngoguey42) [reply src/irmin/tree_intf.ml:375]: 👍 

Opened an issue here https://github.com/mirage/irmin/issues/1759

**Comment by Ngoguey42** [comment]
Let's merge when CI is green

**Pull request merged**